### PR TITLE
Treat all SQL exceptions the same in DatabaseErrorPageMiddlewareTest

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

File: test/Diagnostics.EFCore.FunctionalTests/DatabaseErrorPageMiddlewareTest.cs [hunk 1]
```diff
@@ -322,7 +322,20 @@ public ApplyMigrationsMiddleware(RequestDelegate next)
             var server = new TestServer(builder);
 
             var ex = await Assert.ThrowsAsync<SqlException>(async () =>
-                await server.CreateClient().GetAsync("http://localhost/"));
+            {
+                try
+                {
+                    await server.CreateClient().GetAsync("http://localhost/");
+                }
+                catch (InvalidOperationException ope)
```
Comment (pranavkm) [inline test/Diagnostics.EFCore.FunctionalTests/DatabaseErrorPageMiddlewareTest.cs]:
```
catch (InvalidOperationException exception) when (exception.InnerException != null)
{
  throw exception.InnerException;
}
```

**Comment by pranavkm** [comment]
@Eilon should this go in preview1? Test stabilization and all that.

**Comment by Eilon** [comment]
Have we seen this test fail recently? If so, I'm fine porting this to preview1. @ajcvickers thoughts?

**Comment by Eilon** [comment]
Ah just saw the related bug https://github.com/aspnet/Diagnostics/issues/366#issuecomment-297567879

Definitely looks worth porting to me.

**Comment by ajcvickers** [comment]
@Eilon @AndriySvyryd Looks like we should port it.

**Comment by AndriySvyryd** [comment]
Done

**Pull request merged**